homeassistant/helpers/script.py

```diff
@@ -733,6 +834,22 @@ def __init__(
             for action in self.sequence
         )
 
+        self._repeat_script = {}
+        for step, action in enumerate(sequence):
+            if cv.determine_script_action(action) == cv.SCRIPT_ACTION_REPEAT:
+                step_name = action.get(CONF_ALIAS, f"Repeat at step {step}")
+                sub_script = Script(
+                    hass,
+                    action[CONF_REPEAT][CONF_SEQUENCE],
+                    f"{name}: {step_name}",
+                    script_mode=SCRIPT_MODE_PARALLEL,
```

Comment:
We are going to run all the repeat actions in parallel? I would not expect that.

Reply:
No, the repeat block will be run synchronously within a "run." But, if the parent script is configured in parallel mode, then the repeat block needs to be able to be run in parallel, too. (I.e., if the parent script is run multiple times simultaneously because it is configured for parallel mode, then the "sub script" needs to be able to do that, too.) If the parent is _not_ parallel mode, then it doesn't matter because the parent can only run once at any given time.

Reply:
Ah right. I incorrectly thought that we would create a new script object per run, but we actually share it.

Reply:
Yeah, in Thomas' first stab at implementing the branch/choose action that's what he did, but I decided it was more efficient to create it once, and let it manage the multiple runs (just like in the "parent" case.)